spec/controllers/application_controller/tags_spec.rb

```diff
@@ -113,6 +113,73 @@ def convert_to_region_id(id)
         expect(controller.instance_variable_get(:@tagging)).not_to be_nil
       end
     end
+
+    # I CHALLENGE YOU to make this spec smaller...
+    context "saving changes" do
+      let(:host)            { FactoryBot.create(:host) }
+
+      let(:department)      { FactoryBot.create(:classification_department_with_tags) }
+      let(:environment)     { FactoryBot.create(:classification_environment_with_tags) }
+
+      let(:dept_accounting) { department.entries.find_by_description("Accounting") }
+      let(:dept_finance)    { department.entries.find_by_description("Financial Services") }
+      let(:env_accounting)  { environment.entries.find_by_description("Accounting") }
+      let(:env_production)  { environment.entries.find_by_description("Production") }
+
+      let(:params)          { {"data" => data, "button" => "save", :id => host.id} }
+      let(:s)               { {:tag_db => Host} }
+
+      let(:data) do
+        [
+          {
+            "description" => department.description,
+            "id"          => department.id.to_s,
+            "values"      => [{ "id" => dept_finance.id.to_s, "description" => dept_finance.description }]
+          },
+          {
+            "description" => environment.description,
+            "id"          => environment.id.to_s,
+            "values"      => [{ "id" => env_production.id.to_s, "description" => env_production.description }]
+          }
+        ].to_json
+      end
+
+      before do
+        user = FactoryBot.create(:user, :userid => "for_science")
+        # login_as user # only sets User.current_user
+        controller.current_user = user
+
+        allow(user).to receive(:get_timezone).and_return(Time.zone)
+        allow(controller).to receive(:button_url)
+        allow(controller).to receive(:render)
+        allow(controller).to receive(:javascript_redirect)
+        allow(controller).to receive(:previous_breadcrumb_url)
+        allow(controller).to receive(:load_edit).and_return(true)
+
+        # host.tag_add([dept_accounting.name, env_accounting.name])
```

Comment:
Probably should delete this, but this was my first attempt at assigning tags... it wasn't correct.

Reply:
Yeah this should be removed.